sopel/config/types.py

```diff
@@ -97,18 +101,26 @@ def configure_setting(self, name, prompt, default=NO_DEFAULT):
 
 
 class BaseValidated(object):
-    """The base type for a descriptor in a ``StaticSection``."""
+    """The base type for a setting descriptor in a :class:`StaticSection`
```

Comment:
It's unclear because (I think) there hasn't been one until recently, so stuff is inconsistent everywhere. Personally, I've been working under the guideline that the top-line short description should be a complete sentence, with period (and just missed this one). I'll carry on with that idea when going over (more) stuff for #1565.

The remaining body of a docstring should _definitely_ consist of grammatically complete language, with all the punctuation that implies. 😁